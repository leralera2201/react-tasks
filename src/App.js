import React from 'react';
import { Switch, Route } from 'react-router-dom';

import Header from "@components/Header";

import Buttons from "./pages/Buttons";
import Typography from "./pages/Typography";
import Dropdown from "./pages/Dropdown";
import Select from "./pages/Select";
import Alert from "./pages/Alert";


function App() {
  return (
    <div className="App">
        <Header />
        <Switch>
            <Route path="/" exact component={Buttons}/>
            <Route path="/typography" exact component={Typography}/>
            <Route path="/dropdown" exact component={Dropdown}/>
            <Route path="/select" exact component={Select}/>
            <Route path="/alert" exact component={Alert}/>
        </Switch>
    </div>
  );
}

export default App;
