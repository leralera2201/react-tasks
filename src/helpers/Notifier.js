import React from 'react';
import { toast } from 'react-toastify';

import AlertItem from '@components/Alert';

class Notifier {
  notifySecondary = (message) => {
    toast(<AlertItem color="secondary" message={message} />, {
      hideProgressBar: true,
    });
  };

  notifyError = (message) => {
    toast(<AlertItem color="danger" message={message} />, {
      hideProgressBar: true,
    });
  };

  notifyWarning = (message) => {
    toast(<AlertItem color="warning" message={message} />, {
      hideProgressBar: true,
    });
  };

  notifySuccess = (message) => {
    toast(<AlertItem color="success" message={message} />, {
      hideProgressBar: true,
    });
  };

  notifyInfo = (message) => {
    toast(<AlertItem color="info" message={message} />, {
      hideProgressBar: true,
    });
  };
}

const notifier = new Notifier();

export default notifier;
