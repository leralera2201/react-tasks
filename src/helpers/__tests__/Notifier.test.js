import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

import App from "../../App";

import { notifier } from "../Notifier";

describe("Testing Notifier helpers", () => {
  it("error toast", () => {
    notifier.notifyError("Error");
    const wrapper = shallow(<App />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("success toast", () => {
    notifier.notifySuccess("Success");
    const wrapper = shallow(<App />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("warning toast", () => {
    notifier.notifyWarning("Warning");
    const wrapper = shallow(<App />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("info toast", () => {
    notifier.notifyInfo("Info");
    const wrapper = shallow(<App />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("secondary toast", () => {
    notifier.notifySecondary("Secondary");
    const wrapper = shallow(<App />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
