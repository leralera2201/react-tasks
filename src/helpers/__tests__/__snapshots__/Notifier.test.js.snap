// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Notifier helpers error toast 1`] = `
<Fragment>
  <Header />
  <ToastContainer
    autoClose={5000}
    closeButton={[Function]}
    closeOnClick={true}
    draggable={true}
    draggablePercent={80}
    hideProgressBar={false}
    newestOnTop={false}
    pauseOnFocusLoss={true}
    pauseOnHover={true}
    position="top-right"
    role="alert"
    rtl={false}
    transition={[Function]}
  />
  <div
    className="wrapper"
  >
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "type": [Function],
          }
        }
        exact={true}
        path="/books"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "type": [Function],
          }
        }
        exact={true}
        path="/books/create"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "type": [Function],
          }
        }
        exact={true}
        path="/books/:id"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "type": [Function],
          }
        }
        exact={true}
        path="/books/:id/update"
      />
    </Switch>
  </div>
  <Footer />
</Fragment>
`;

exports[`Testing Notifier helpers info toast 1`] = `
<Fragment>
  <Header />
  <ToastContainer
    autoClose={5000}
    closeButton={[Function]}
    closeOnClick={true}
    draggable={true}
    draggablePercent={80}
    hideProgressBar={false}
    newestOnTop={false}
    pauseOnFocusLoss={true}
    pauseOnHover={true}
    position="top-right"
    role="alert"
    rtl={false}
    transition={[Function]}
  />
  <div
    className="wrapper"
  >
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "type": [Function],
          }
        }
        exact={true}
        path="/books"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "type": [Function],
          }
        }
        exact={true}
        path="/books/create"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "type": [Function],
          }
        }
        exact={true}
        path="/books/:id"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "type": [Function],
          }
        }
        exact={true}
        path="/books/:id/update"
      />
    </Switch>
  </div>
  <Footer />
</Fragment>
`;

exports[`Testing Notifier helpers secondary toast 1`] = `
<Fragment>
  <Header />
  <ToastContainer
    autoClose={5000}
    closeButton={[Function]}
    closeOnClick={true}
    draggable={true}
    draggablePercent={80}
    hideProgressBar={false}
    newestOnTop={false}
    pauseOnFocusLoss={true}
    pauseOnHover={true}
    position="top-right"
    role="alert"
    rtl={false}
    transition={[Function]}
  />
  <div
    className="wrapper"
  >
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "type": [Function],
          }
        }
        exact={true}
        path="/books"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "type": [Function],
          }
        }
        exact={true}
        path="/books/create"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "type": [Function],
          }
        }
        exact={true}
        path="/books/:id"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "type": [Function],
          }
        }
        exact={true}
        path="/books/:id/update"
      />
    </Switch>
  </div>
  <Footer />
</Fragment>
`;

exports[`Testing Notifier helpers success toast 1`] = `
<Fragment>
  <Header />
  <ToastContainer
    autoClose={5000}
    closeButton={[Function]}
    closeOnClick={true}
    draggable={true}
    draggablePercent={80}
    hideProgressBar={false}
    newestOnTop={false}
    pauseOnFocusLoss={true}
    pauseOnHover={true}
    position="top-right"
    role="alert"
    rtl={false}
    transition={[Function]}
  />
  <div
    className="wrapper"
  >
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "type": [Function],
          }
        }
        exact={true}
        path="/books"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "type": [Function],
          }
        }
        exact={true}
        path="/books/create"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "type": [Function],
          }
        }
        exact={true}
        path="/books/:id"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "type": [Function],
          }
        }
        exact={true}
        path="/books/:id/update"
      />
    </Switch>
  </div>
  <Footer />
</Fragment>
`;

exports[`Testing Notifier helpers warning toast 1`] = `
<Fragment>
  <Header />
  <ToastContainer
    autoClose={5000}
    closeButton={[Function]}
    closeOnClick={true}
    draggable={true}
    draggablePercent={80}
    hideProgressBar={false}
    newestOnTop={false}
    pauseOnFocusLoss={true}
    pauseOnHover={true}
    position="top-right"
    role="alert"
    rtl={false}
    transition={[Function]}
  />
  <div
    className="wrapper"
  >
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "type": [Function],
          }
        }
        exact={true}
        path="/books"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "type": [Function],
          }
        }
        exact={true}
        path="/books/create"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "type": [Function],
          }
        }
        exact={true}
        path="/books/:id"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "type": [Function],
          }
        }
        exact={true}
        path="/books/:id/update"
      />
    </Switch>
  </div>
  <Footer />
</Fragment>
`;
